packages/aws-cdk-lib/core/lib/cfn-fn.ts

```diff
@@ -334,6 +334,10 @@ export class Fn {
     if (conditions.length === 1) {
       return conditions[0] as ICfnRuleConditionExpression;
     }
+    // prevent the error "Fn::Or object requires a list of at least 2" when the condition is split into groups of 10 and 1
```

Comment:
Just a matter of preference, but this seems more readable:

```ts
public static conditionOr(...conditions: ICfnConditionExpression[]): ICfnRuleConditionExpression {
  if (conditions.length === 0) {
    throw new Error('Fn.conditionOr() needs at least one argument');
  }
  if (conditions.length === 1) {
    return conditions[0] as ICfnRuleConditionExpression;
  }
  if (conditions.length <= 10) {
    return new FnOr(...conditions);
  }
  return Fn.conditionOr(..._inGroupsOf(conditions, 10).map(group => Fn.conditionOr(...group)));
}
```

The three base cases are stated explicitly, falling back to the recursive case at the end.